src/main/java/se/sundsvall/emailreader/service/EmailService.java

```diff
@@ -95,7 +96,7 @@ public void sendReport() {
 
 	@Transactional
 	public void saveAndMoveEmail(final Email email, final String emailAddress, final CredentialsEntity credential) throws Exception {
-		emailRepository.save(EmailMapper.toEmailEntity(email, credential.getNamespace(), credential.getMunicipalityId(), credential.getMetadata()));
+		emailRepository.save(EmailMapper.toEmailEntity(email, credential.getNamespace(), credential.getMunicipalityId(), new HashMap<>(credential.getMetadata())));
```

Comment:
Is it necessary to create a new HashMap instead of just passing the credential.getMetadata() map?

Reply:
Yes. The bug was that the reference was shared between the entities. They must be separate (different objects).